### PR TITLE
Implement all digest classes to read the digest.json into a metamodel.

### PR DESCRIPTION
**Pull request by azabbasi**
This PR includes all metadata types that can be read from the digest file. 
The highest level of hierarchy is the MetamodelDigest and it trickles down from there.
The objective of this PR is to feed the MetamodelDigest object (the root) the full digest text and expect the full hierarchy of digest models.
Later on, we will use these digest models to generate the DTDL object model.

### CODE REVIEW

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaEnum.java [hunk 1]
```diff
@@ -14,10 +14,10 @@
  * Generator for java enum
  */
 public class JavaEnum extends JavaDeclaration implements JavaFile {
-    private static ClientLogger logger = new ClientLogger(JavaEnum.class);
-    private List<EnumVal> enumValues;
-    private boolean isSorted;
-    private String typeName;
+    private static final ClientLogger LOGGER = new ClientLogger(JavaEnum.class);
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaEnum.java:17]: why change to upper case?

Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaEnum.java:17]: I made it static final and static final fields need to be all upercase.

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaType.java [hunk 2]
```diff
@@ -16,16 +16,16 @@
  */
 public class JavaType extends JavaDeclaration implements JavaFile {
 
-    private ClientLogger logger = new ClientLogger(JavaType.class);
-    private String typeName;
+    private final ClientLogger logger = new ClientLogger(JavaType.class);
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaType.java:19]: does this need to be static as well?

Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaType.java:19]: yap, I will change them all. good catch.

**Pull request merged**